crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -20,25 +26,41 @@ pub use {auction::Auction, solution::Solution, transaction::Transaction};
 #[derive(Debug)]
 pub struct Settlement {
     solution: Solution,
+    transaction: Transaction,
```

Comment:
I missed this. If we are adding the transaction here, can't we get rid of the solution field? It can now be computed on the fly (as long as we also have the domain separator somehow) and we avoid storing redundant information.